### ISSUE TITLE
How to Train Your Dragon: ocument how to train and use a custom model.

### ISSUE THREAD

**Issue by tnn1t1s** [tnn1t1s/NeurIPS202#5]
Write a doc that shows how to train a model with your own source material and use that model in the magenta/VMPK/fluidsynth setup. 

For training data, use lead lines from New Order, Joy Division, Blink 182 and Goog Charlotte. It’s OK if model is overfit and doesn’t have enough training data. It will be funny.

**Comment by tnn1t1s** [comment]
This is the first step:
     - https://github.com/tensorflow/magenta/blob/master/magenta/scripts/README.md

**Comment by tnn1t1s** [comment]
As a trivial exercise, I started working with the MIDI file available for download here:
     - https://www.youtube.com/watch?v=ON-qtZe0GYY

These are some basic EDM leads that should be interesting enough for training of the basic melody_rnn. so far, I've done the following:

1) opened the midi file in Live and listened.
2) noticed that some of the loops are 8 bars, some are 4 bars, and some are just awful and/or nonsense 
3) normalized all phrases to 8 bars and consolidated into a single midi file
4) use the mido library in python to import the consolidated file and export individual files containing single track, 8 bar leads.

**Comment by tnn1t1s** [comment]
ok - training set is now built . 30 8 bar lead-lines from 2017's 30 most popular EDM bangers. I don't suspect this is nearly enough data but i'm interested in seeing what happens and I want a set that I can reason about easily and be musically familiar with. (although it is a form of torture to listen to these clips)

**Comment by tnn1t1s** [comment]
so the script that converts midi sequences to training data has opinionated view of what constitutes good training data; it kicked out some of my midi files for lacking note variation. :) 

but seriously - this is one example of how the tooling you build can drastically impact the models. For EDM, we may want to overweight sequencer that are interesting but have little note variation. (certainly this is the case for the models i want to build for strangeloop)

https://www.youtube.com/watch?v=tajdHgg6TF8

**Comment by tnn1t1s** [comment]
this model works remarkably well. the limited training set seems to be overfit nicely to the sort of syncopated sequences you hear in EDM and the variation in the examples is low enough that the responses generated are all very similar, with some small variation, and copying the input line. it sounds less like a neural network and more like a broken midi loop pedal. 

next step is to write this up , polish up the scripts I wrote and commit. then finally, record some examples and discuss.

**Comment by tnn1t1s** [comment]
if the model is working as hypothesized, i'd expect that if i play 'mary had a little lamb' as a starting vector, i will get something that sounds like a syncopated EDM lead line version of mary had a little lamb as response. which i do. so yeah.
